compiler/src/dmd/parse.d

```diff
@@ -8730,7 +8730,7 @@ class Parser(AST, Lexer = dmd.lexer.Lexer) : Lexer
                 else
                 {
                     AST.Type t = parseType(); // cast( type )
-                    t = t.addMod(m); // cast( const type )
+                    t = t.addSTC(AST.ModToStc(m)); // cast( const type )
```

Comment:
addMod also merges the newly created type. addSTC only stores the STC which is then merged during type semantic.